src/targets/symbolCollector.ts

```diff
@@ -0,0 +1,108 @@
+import { BaseArtifactProvider } from '../artifact_providers/base';
+import { TargetConfig } from '../schemas/project_config';
+import { ConfigurationError } from '../utils/errors';
+import { BaseTarget } from './base';
+import { withTempDir } from '../utils/files';
+import { promises as fsPromises } from 'fs';
+import { checkExecutableIsPresent, spawnProcess } from '../utils/system';
+import { join } from 'path';
+
+const DEFAULT_SYM_COLLECTOR_SERVER_ENDPOINT =
+  'https://symbol-collector.services.sentry.io/';
+/**
+ * Name of the binary of the symbol collector.
+ * Must be available in the path.
+ */
+export const SYM_COLLECTOR_BIN_NAME = 'SymbolCollector.Console';
+
+/** Config options for the "symbol-collector" target. */
+interface SymbolCollectorTargetConfig {
+  /** Server endpoint to upload symbols. */
+  serverEndpoint: string;
+  /** batch-type of the symbols to be uploaded. */
+  batchType: string;
+  /** Prefix of the bundle ID to be uploaded. */
+  bundleIdPrefix: string;
+}
+
+export class SymbolCollector extends BaseTarget {
+  /** Target name */
+  public readonly name: string = 'symbol-collector';
+  /** Target options */
+  public readonly symbolCollectorConfig: SymbolCollectorTargetConfig;
+
+  public constructor(
+    config: TargetConfig,
+    artifactProvider: BaseArtifactProvider
+  ) {
+    super(config, artifactProvider);
+    this.symbolCollectorConfig = this.getSymbolCollectorConfig();
+  }
+
+  private getSymbolCollectorConfig(): SymbolCollectorTargetConfig {
+    // The Symbol Collector should be available in the path
+    checkExecutableIsPresent(SYM_COLLECTOR_BIN_NAME);
```

Comment:
This is not really checking a config unless you make the bin name configurable through an env variable or the config file itself.